contrib/mac/framework/Makefile

```diff
@@ -0,0 +1,146 @@
+$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).$(SOMAJOR).dylib: | $(framework_destdirs)
+	ln -s -f ../$(FRAMEWORK_NAME) $@
+$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).dylib: | $(framework_destdirs)
+	ln -s -f ../$(FRAMEWORK_NAME) $@
+$(DESTDIR)$(prefix)/$(framework_currver)/libexec: | $(framework_destdirs)
+	ln -s -f Helpers $@
+$(DESTDIR)$(prefix)/$(framework_currver)/share: | $(framework_destdirs)
+	ln -s -f Resources $@
+$(DESTDIR)$(prefix)/$(framework_currver)/include: | $(framework_destdirs)
+	ln -s -f Headers $@
+$(DESTDIR)$(prefix)/$(framework_currver)/etc: | $(framework_destdirs)
+	ln -s -f Resources $@
+
+hier_symlinks: \
+	$(DESTDIR)$(prefix)/$(framework_currver)/bin \
+	$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).$(SOMAJOR).$(SOMINOR).dylib \
+	$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).$(SOMAJOR).dylib \
+	$(DESTDIR)$(prefix)/$(framework_currver)/lib/libjulia$(JULIA_LIBSUFFIX).dylib \
+	$(DESTDIR)$(prefix)/$(framework_currver)/libexec \
+	$(DESTDIR)$(prefix)/$(framework_currver)/share \
+	$(DESTDIR)$(prefix)/$(framework_currver)/include \
+	$(DESTDIR)$(prefix)/$(framework_currver)/etc
+
+$(DESTDIR)$(prefix)/$(framework_infoplist): | $(framework_destdirs)
+	/usr/libexec/PlistBuddy -x -c "Clear dict" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleName string $(FRAMEWORK_NAME)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleExecutable string $(FRAMEWORK_NAME)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleIdentifier string $(DARWIN_CODESIGN_ID_BASE).lib" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleVersion string $(JULIA_VERSION_OPT_COMMIT)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleShortVersionString string $(JULIA_MAJOR_VERSION).$(JULIA_MINOR_VERSION).$(JULIA_PATCH_VERSION)" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleSignature string ???" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundlePackageType string FMWK" $@
+	/usr/libexec/PlistBuddy -x -c "Add :CFBundleInfoDictionaryVersion string 6.0" $@
+	/usr/libexec/PlistBuddy -x -c "Add :NSHumanReadableCopyright string \"Copyright © 2009-2018 Julia project contributors (https://github.com/JuliaLang/julia/contributors). See LICENSE.md.\"" $@
+
+toplevelinstall:
+ifneq ($(DARWIN_FRAMEWORK),1)
+	$(error Darwin framework is not enabled. Please set DARWIN_FRAMEWORK=1)
+endif
+	$(MAKE) -C $(BUILDROOT) install
+
+framework: toplevelinstall $(DESTDIR)$(prefix)/$(framework_infoplist) | $(framework_current_symlinks) $(framework_version_symlink) $(framework_destdirs) hier_symlinks
+	
+	# retain correct release/debug julia and fix paths
+	mv $(DESTDIR)$(bindir)/$(notdir $(JULIA_EXECUTABLE)) $(DESTDIR)$(bindir)/julia
+	-rm $(DESTDIR)$(bindir)/julia-debug
+	$(INSTALL_NAME_CHANGE_CMD) @rpath/libjulia$(JULIA_LIBSUFFIX).dylib @rpath/$(FRAMEWORK_NAME) $(DESTDIR)$(bindir)/julia
+	$(JULIAHOME)/contrib/mac/framework/delete-all-rpaths.sh $(DESTDIR)$(bindir)/julia
```

Comment:
Why do you need to delete all the rpaths?

Reply:
The rpaths need to change due to the different layout of a framework. It's easier to delete all rpaths and set only the correct needed rpaths.  Otherwise, the non-framework incorrect rpaths need to be precisely known before they can be removed.